base/stat.jl

```diff
@@ -126,6 +189,50 @@ lstat(path...) = lstat(joinpath(path...))
 Equivalent to `stat(file).mode`.
 """
 filemode(st::StatStruct) = st.mode
+filemode_string(st::StatStruct) = filemode_string(st.mode)
+function filemode_string(mode)
+    filemode_table = (
+        ((S_IFLNK,        "l"),
+        (S_IFSOCK,        "s"),  # Must appear before IFREG and IFDIR as IFSOCK == IFREG | IFDIR
+        (S_IFREG,         "-"),
+        (S_IFBLK,         "b"),
+        (S_IFDIR,         "d"),
+        (S_IFCHR,         "c"),
+        (S_IFIFO,         "p")),
+
+        ((S_IRUSR,         "r"),),
+        ((S_IWUSR,         "w"),),
```

Comment:
Why is the formatting of some of these different?

It is typically better to put these constant tables outside of the function. While the compiler may be able to pre-compute them, it is generally better not to rely upon that.

Reply:
Moved out and reformatted